packages/core/src/App/Containers/Redirect/redirect.jsx

```diff
@@ -34,28 +33,31 @@ const Redirect = ({
 
     switch (action_param) {
         case 'signup': {
-            if (is_appstore) {
-                // TODO: redirect
-                // history.push({
-                //     pathname: routes.dashboard,
-                //     search: url_query_string,
-                // });
-                // redirected_to_route = true;
-            } else if (is_eu_country || is_eu) {
-                history.push({
-                    pathname: routes.trading_hub,
-                    search: url_query_string,
-                });
-                redirected_to_route = true;
-            } else {
-                history.push({
-                    pathname: routes.onboarding,
-                    search: url_query_string,
-                });
-                redirected_to_route = true;
-            }
-            sessionStorage.removeItem('redirect_url');
-            toggleAccountSignupModal(true);
+            WS.wait('website_status').then(() => {
+                if (is_appstore) {
+                    // TODO: redirect
+                    // history.push({
+                    //     pathname: routes.dashboard,
+                    //     search: url_query_string,
+                    // });
+                    // redirected_to_route = true;
+                } else if (isEuropeCountry()) {
+                    setIsPreAppStore(true);
+                    history.push({
+                        pathname: routes.trading_hub,
+                        search: url_query_string,
+                    });
+                    redirected_to_route = true;
+                } else {
+                    history.push({
+                        pathname: routes.onboarding,
+                        search: url_query_string,
+                    });
```

Comment:
since this is same for both conditions, move it to line 58

Reply:
Alrights